### PR TITLE
Upgrades the coat racks and adds security labcoats

### PR DESCRIPTION
**Pull request by Casper3667**
This PR turned out slightly bigger than expected. It does the following:
- The coat rack can now carry forensic coats and the detective jacket.
- The coat rack can now carry some hats alongside the coats. The coat is removed first, then the hat.
- A security labcoat was added and replaces the normal labcoat in the investigator locker.
- While in the file, I removed several mentions of labcoats having better protection as they did not, and a few who had ARMOR_BIO_RESISTANT armor because the base already has that. Nothing was mechanically changed about the labcoats.

### CODE REVIEW

**Comment by Casper3667** [comment]
!review

**Review by Geevies — APPROVED** [review]

File: code/game/objects/structures/coathanger.dm [hunk 1]
```diff
@@ -20,31 +23,53 @@
 		user.put_in_hands(coat)
 		coat = null
 		update_icon()
+	else if(hat)
+		user.visible_message("[user] takes [hat] off \the [src].", "You take [hat] off the \the [src]")
+		hat.forceMove(get_turf(user))
```
Comment (Geevies) [inline code/game/objects/structures/coathanger.dm:28]: put_in_hands already force_moves the thing

File: code/game/objects/structures/coathanger.dm [hunk 2]
```diff
@@ -20,31 +23,53 @@
 		user.put_in_hands(coat)
 		coat = null
 		update_icon()
+	else if(hat)
+		user.visible_message("[user] takes [hat] off \the [src].", "You take [hat] off the \the [src]")
```
Comment (Geevies) [inline code/game/objects/structures/coathanger.dm:27]:
```suggestion
		user.visible_message("[user] takes [hat] off \the [src].", SPAN_NOTICE("You take [hat] off the \the [src]."))
```

File: code/game/objects/structures/coathanger.dm [hunk 3]
```diff
@@ -20,31 +23,53 @@
...
-	if(is_type_in_list(W, allowed))
-		can_hang = 1
-	if (can_hang && !coat)
+	var/can_hang_coat = 0
+	var/can_hang_hat = 0
+	if(is_type_in_list(W, allowed_coats))
+		can_hang_coat = 1
+	else if(is_type_in_list(W, allowed_hats))
+		can_hang_hat = 1
+	if (can_hang_coat && !coat)
```
Comment (Geevies) [inline code/game/objects/structures/coathanger.dm:40]: can't you just replace can_hang_coat with the is_type_in_list check

File: code/game/objects/structures/coathanger.dm [hunk 4]
```diff
@@ -20,31 +23,53 @@
...
 		user.visible_message("[user] hangs [W] on \the [src].", "You hang [W] on the \the [src]")
 		coat = W
 		user.drop_from_inventory(coat, src)
 		update_icon()
+	else if (can_hang_hat && !hat)
+		user.visible_message("[user] hangs [W] on \the [src].", "You hang [W] on the \the [src]")
```
Comment (Geevies) [inline code/game/objects/structures/coathanger.dm:46]:
```suggestion
		user.visible_message("[user] hangs [W] on \the [src].", SPAN_NOTICE("You hang [W] on the \the [src]."))
```

File: code/game/objects/structures/coathanger.dm [hunk 5]
```diff
@@ -20,31 +23,53 @@
...
-		can_hang = 1
-	if (can_hang && !coat)
+	var/can_hang_coat = 0
+	var/can_hang_hat = 0
+	if(is_type_in_list(W, allowed_coats))
+		can_hang_coat = 1
+	else if(is_type_in_list(W, allowed_hats))
+		can_hang_hat = 1
+	if (can_hang_coat && !coat)
 		user.visible_message("[user] hangs [W] on \the [src].", "You hang [W] on the \the [src]")
```
Comment (Geevies) [inline code/game/objects/structures/coathanger.dm:41]:
```suggestion
		user.visible_message("[user] hangs [W] on \the [src].", SPAN_NOTICE("You hang [W] on the \the [src]."))
```

File: code/game/objects/structures/coathanger.dm [hunk 6]
```diff
@@ -20,31 +23,53 @@
...
-		can_hang = 1
-
-	if (can_hang && !coat)
+	var/can_hang_coat = 0
+	var/can_hang_hat = 0
+	if(is_type_in_list(mover, allowed_coats))
+		can_hang_coat = 1
+	else if(is_type_in_list(mover, allowed_hats))
+		can_hang_hat = 1
+	if (can_hang_coat && !coat)
```
Comment (Geevies) [inline code/game/objects/structures/coathanger.dm:61]: replace can_hang_coat with the istypeinlist check

**Review by Alberyk — APPROVED** [review]

**Pull request merged**